src/testbase.rs

```diff
@@ -0,0 +1,329 @@
+//! Common utilities shared by unit tests.
+#![cfg(test)]
+
+use std::collections::HashSet;
+
+use nalgebra::{Point3, Vector3};
+
+use aabb::{AABB, Bounded};
+use bounding_hierarchy::{BoundingHierarchy, BHShape};
+use ray::Ray;
+
+/// A vector represented as a tuple
+pub type TupleVec = (f32, f32, f32);
+
+/// Convert a `TupleVec` to a `nalgebra` point.
+pub fn tuple_to_point(tpl: &TupleVec) -> Point3<f32> {
+    Point3::new(tpl.0, tpl.1, tpl.2)
+}
+
+/// Convert a `TupleVec` to a `nalgebra` vector.
+pub fn tuple_to_vector(tpl: &TupleVec) -> Vector3<f32> {
+    Vector3::new(tpl.0, tpl.1, tpl.2)
+}
+
+/// Define some `Bounded` structure.
+pub struct UnitBox {
+    pub id: i32,
+    pub pos: Point3<f32>,
+    node_index: usize,
+    pub has_good_bh_position: bool, // If the XBox's node in the BVH gets moved, we panic if this is true
```

Comment:
Updated in 2a164b5.
The ticks around a struct name only appear in doc comments, not regular comments throughout the whole code, so I didn't change that.